actions/lib/action.py

```diff
@@ -59,8 +61,43 @@ def __init__(self, config):
         if region:
             self.credentials['region'] = region
 
+        self.session = Session(aws_access_key_id=self.credentials['aws_access_key_id'],
+                               aws_secret_access_key=self.credentials['aws_secret_access_key'])
+
+        self.account_id = self.session.client('sts').get_caller_identity().get('Account')
+        self.cross_roles_arns = {
+            arn.split(':')[4]: arn for arn in self.config.get('actions', {}).get('roles', [])
+        }
+
         self.resultsets = ResultSets()
 
+    def assume_role(self, account_id):
+        ''' Assumes role and setup Boto3 session for the cross-account capability'''
+        if account_id == self.account_id:
+            return
+
+        try:
+            assumed_role = self.session.client('sts').assume_role(
+                RoleArn=self.cross_roles_arns[account_id],
+                RoleSessionName='StackStormEvents'
+            )
+        except ClientError:
+            self.logger.error('Failed to assume role as account %s when using the AWS session \
+                               client. Check the roles configured for the AWS pack and ensure \
+                               that the \'%s\' is still valid. ',
```

Comment:
This is Python, please write "Pythonicly". Instead of using single quotes and escaping them everywhere, I would use surrounding double quotes, remove line continuations, and remove the space character at the end of the log message:

```python
            self.logger.error("Failed to assume role as account %s when using the AWS session "
                              "client. Check the roles configured for the AWS pack and ensure "
                              "that the '%s' is still valid.",
```